app/models/miq_queue.rb

```diff
@@ -363,7 +363,7 @@ def deliver(requester = nil)
       message = "Expired on [#{expires_on}]"
       _log.error("#{MiqQueue.format_short_log_msg(self)}, #{message}")
       status = STATUS_EXPIRED
-    rescue Exception => error
+    rescue StandardError, SyntaxError => error
```

Comment:
@kbrock Nice.  Thanks for tackling this.

So, yes, I agree with @NickLaMuro and you here.  This is clearly the right thing to do.   Going forward, we really need to remove these rescue exceptions and I'd prefer we fix the edge cases where non-standard error exceptions are raised and the worker dies.

@kbrock One suggested change is to put the commit `49e66c99444aff00` in the commit message and mention that it was solved wrong:  handling a `SyntaxError` by rescuing `Exception`.  The latter part is really why you're undoing this change.